### PR TITLE
extend graph navigation - allow for graph to be maintained during node expansion/navigation

### PR DESCRIPTION
**Pull request by davemasselink**
allow for graph to be maintained while user requested additions enter the dataset
* modifying GraphContainer such that it uses the redux store rather than component local state
* lots of tweaks to graph component and related utils due to change in data flow
* updating async spinner to use data from central store, as well

video example: https://www.useloom.com/share/bb0a7f160db24987b9ad7f2ecbc35ff9

### CODE REVIEW

**Comment by fzxu** [comment]
@davemasselink could you rebase the PR from the latest master?

File: app/src/actions/entityActions.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  return dispatch => {
+    return ApiService.getEntity(uid)
+      //TODO:DM - can this hard-coded idx be cleaned up?
+      .then(json => dispatch(receiveEntity(json.objects[0])));
+  };
+};
+
+export const fetchEntities = uids => {
+  return dispatch => {
+    return Promise.all(uids.map(uid =>
```
Comment (fzxu) [inline app/src/actions/entityActions.js]: question: is `Promise` needed here?

File: app/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
 FROM node:8.12.0 as build-stage
```
[the remark this answers is not in the document]
Comment (davemasselink) [reply app/Dockerfile]: @fzxu, fortunately... all of nodejs is NOT in the runtime image

take a quick look at the comment starting on line 10 to see how the multiple stages of this dockerfile are used, and let me know if you feel that the comment could be extended or re-worded to clarify

further, I've confirmed that the artifactory images are on the order of 25MB which I know they could not be were they including nodejs

File: app/src/actions/entityActions.js (hunk 1 above)
Comment (davemasselink) [reply app/src/actions/entityActions.js]: no, I believe not... thanks for asking

at one point I had had a `.then(...)` chaining off of the `Promise.all()`method, which is a method that allows you to combine multiple promises into one, but am no longer using the follow-up clause

i'll confirm that the code works correctly without the extra layer of wrapping all promises returned by the `dispatch()` calls into one that is returned from the thunk func

**Comment by davemasselink** [comment]
@fzxu, yes... I'll re-base to make sure it'll merge cleanly with `master`

and make a couple tweaks based on your comments and another issue I discovered

File: app/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (fzxu) [reply app/Dockerfile]: ah. I got you. the 2nd stage got collapsed when reviewing...

**Comment by fzxu** [comment]
LGTM. feel free to merge it yourself if you are not making additional changes to this PR.

**Comment by davemasselink** [comment]
@fzxu, thx... yeah, I'll merge once I've pushed my final changes and all checks are still passing

**Pull request merged**